codespell_lib/_codespell.py

```diff
@@ -95,8 +95,10 @@ class QuietLevels(object):
 class GlobMatch(object):
     def __init__(self, pattern):
         if pattern:
-            # Pattern might be a list of comma-delimited strings
-            self.pattern_list = ','.join(pattern).split(',')
+            # Pattern might be a list of comma-separated patterns,
```

Comment:
Yes, I agree about the use case of a file name including a comma. However, this pull request doesn't alter the behaviour of the `-S foo,bar` option, does it?

Reply:
No, sorry I'm basically reviewing the code when I'm reviewing your changes, given it's unlikely a TODO in here saying this will break with files with a comma in is sufficient.

Reply:
Since the pattern separator is a comma, I think it's obvious we cannot handle patterns containing a comma. Perhaps we should make that clear in the documentation:
_Because the comma is a pattern separator, patterns cannot include commas. In the unlikely event you need to match a path containing a comma, please consider working around this limitation using globs._

Reply:
Or we could introduce an escape mechanism, perhaps `\,`. But I'd wait for actual users to ask for it...